spec/factories/collection_types.rb

```diff
@@ -110,5 +80,60 @@
 
   factory :admin_set_collection_type, class: Hyrax::CollectionType do
     initialize_with { Hyrax::CollectionType.find_or_create_admin_set_type }
+
+    transient do
+      creator_user { nil }
+      creator_group { nil }
+      manager_user { nil }
+      manager_group { nil }
+    end
+
+    after(:create) do |collection_type, evaluator|
+      CollectionTypeFactoryHelper.process_access(collection_type, evaluator)
+    end
+  end
+
+  class CollectionTypeFactoryHelper
+    def self.process_access(collection_type, evaluator) # rubocop:disable Metrics/MethodLength
```

Comment:
Moved access grants to the helper class to allow the code to be shared among factories.